### PR TITLE
[RW-285] Define clone API

### PR DESCRIPTION
**Pull request by calbach**
Background: https://docs.google.com/document/d/1RfTkoceNYL0l76qGWX9RLlKdPcI6rNdGywkHtzqvpZU

CC @danqrodney

### CODE REVIEW

File: api/src/main/resources/workbench.yaml [hunk 1]
```diff
@@ -486,6 +486,37 @@ paths:
           schema:
             $ref: "#/definitions/ShareWorkspaceResponse"
 
+  /api/v1/workspaces/{workspaceNamespace}/{workspaceId}/clone:
+    post:
+      tags:
+        - workspaces
+      description: >
+        Clone an existing workspace, with optional modifications to workspace metadata. Caller will
```
Comment (jmandel) [inline api/src/main/resources/workbench.yaml]: Clarifying "optional modification" -- does this just refer to the description field?

File: api/src/main/resources/workbench.yaml [hunk 2]
```diff
@@ -1592,6 +1623,31 @@ definitions:
...
+     - workspace
+    properties:
+      workspace:
+        $ref: "#/definitions/Workspace"
+        description: >
+          Workspace metadata to be applied to the cloned workspace upon creation. The following
+          fields are accepted:
+            - name (required)
+            - namespace (required)
+            - researchPurpose (required)
```
Comment (jmandel) [inline api/src/main/resources/workbench.yaml]: Does this contemplate the research purpose as ... a string? Or a full structured purpose as we collect when creating a workspace from scratch? Assuming the latter, this works great (will be good to reuse the structure for research purpose  if we can, i.e. the `ResearchPurpose` object from the this swagger file).

File: api/src/main/resources/workbench.yaml (hunk 1 above)
Comment (calbach) [reply api/src/main/resources/workbench.yaml]: Yeah, actually it's not really optional in that several fields are required. Reworded this.

File: api/src/main/resources/workbench.yaml (hunk 2 above)
Comment (calbach) [reply api/src/main/resources/workbench.yaml]: The full structure. These refer to fields on the `Workspace` object, which includes a structured `researchPurpose` field. Reworded the above document to make this clearer.

Note: only a subset of the fields on the research purpose will actually be settable, e.g. `approvalTime` will be ignored, following the same rules `CreateWorkspace` should (but by my reading - actually doesn't currently, which I will fix).

**Pull request merged**